org.eclipse.debug.core/core/org/eclipse/debug/internal/core/LaunchConfigurationWorkingCopy.java

```diff
@@ -271,16 +275,33 @@ private void doSave0(IProgressMonitor monitor) throws CoreException {
 		}
 	}
 
+	private static String getLineDelimiterPreference(IFile file) {
+		IScopeContext[] scopeContext;
+		if (file != null && file.getProject() != null) {
+			// project preference
+			scopeContext = new IScopeContext[] {
+					new ProjectScope(file.getProject()) };
+			String lineDelimiter = Platform.getPreferencesService().getString(Platform.PI_RUNTIME, Platform.PREF_LINE_SEPARATOR, null, scopeContext);
+			if (lineDelimiter != null) {
+				return lineDelimiter;
+			}
+		}
+		// workspace preference
+		scopeContext = new IScopeContext[] { InstanceScope.INSTANCE };
+		return Platform.getPreferencesService().getString(Platform.PI_RUNTIME, Platform.PREF_LINE_SEPARATOR, System.lineSeparator(), scopeContext);
+	}
+
 	/**
 	 * Writes the new configuration information to a file.
 	 * @param monitor the {@link IProgressMonitor}
 	 *
 	 * @exception CoreException if writing the file fails
 	 */
 	protected void writeNewFile(IProgressMonitor monitor) throws CoreException {
+		String lineDelimeter = getLineDelimiterPreference(getFile());
 		String xml = null;
 		try {
-			xml = getInfo().getAsXML();
+			xml = getInfo().getAsXML(lineDelimeter);
```

Comment:
This is where we write to a new file that the user specified. Here, since the destination is something the user is aware of and lives in the regular file system, we choose to honor the newline preference and get the line delimiter from the configuration settings (for the specific target file).